src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs

```diff
@@ -123,7 +123,8 @@ private async Task<bool> VerifyAsync(string responseToken)
                     { "response", responseToken }
                 });
 
-                var response = await _httpClient.PostAsync($"{_reCaptchaSettings.ReCaptchaApiUri.TrimEnd('/')}/siteverify", content);
+                var httpClient = _httpClientFactory.CreateClient(nameof(ReCaptchaService));
```

Comment:
Maybe `_httpClient ??= ...` in case it has been already created in the same scope.

Reply:
@jtkech that's is handled by the Factory. `_httpClient ??=` will create a singleton instance on the top of the factory.

Reply:
https://github.com/OrchardCMS/OrchardCore/blob/e228e452311fe16e74eaddcee1376c8b9662df8f/src/OrchardCore/OrchardCore/Modules/Overrides/HttpClient/TenantHttpClientFactory.cs#L118-L133

Reply:
Notice the `ThrowIfNull` helper.

Reply:
@jtkech i'll change it. I still don't think this is needed. the service is registered transient.

Reply:
Maybe not so important, will look at it.

Just saw that I was doing the same ;)

No it is now a scoped service

            services.AddScoped<ReCaptchaService>()

Reply:
Look at the PR changes. I changed it to transient

Reply:
@jtkech I changed it back to scoped and saved the httpClients instances into a local variable to allow reusing it.

Reply:
Let `ReCaptchaService` scoped so that it is realized once per request.

For simplicity let it as you first did, would need to look at it again but creating a `new HttpClient()` is not a so important problem, anyway looks like it is used at most once in a given request.

Reply:
I missed your last comment, okay cool then

Reply:
@jtkech not it is set to this `_httpClient ??= _httpClientFactory.CreateClient(nameof(ReCaptchaService));`

and I registered it as scoped. `services.AddScoped<ReCaptchaService>()`

Reply:
Yes this is what I understood and I was suggesting, so all good

Reply:
@jtkech cool. If you have no additional request, please approve or merge it.